GeoLib/PointVec.cpp

```diff
@@ -41,33 +41,48 @@ PointVec::PointVec (const std::string& name, std::vector<Point*>* points,
 		}
 	}
 
+	std::vector<std::size_t> rm_pos;
 	// add all points in the oct tree in order to make them unique
 	_pnt_id_map.resize(number_of_all_input_pnts);
 	std::iota(_pnt_id_map.begin(), _pnt_id_map.end(), 0);
 	GeoLib::Point * ret_pnt(nullptr);
-	std::size_t cnt(0);
 	for (std::size_t k(0); k<_data_vec->size(); ++k) {
 		GeoLib::Point *const pnt((*_data_vec)[k]);
 		if (! _oct_tree->addPoint(pnt, ret_pnt)) {
 			_pnt_id_map[pnt->getID()] = ret_pnt->getID();
+			rm_pos.push_back(k);
 			delete (*_data_vec)[k];
 			(*_data_vec)[k] = nullptr;
 		} else {
-			_pnt_id_map[k] = cnt;
-			cnt++;
+			_pnt_id_map[k] = pnt->getID();
 		}
 	}
 
 	auto const data_vec_end = std::remove(_data_vec->begin(), _data_vec->end(), nullptr);
 	_data_vec->erase(data_vec_end, _data_vec->end());
 
-	// set value of the point id to the position of the point within _data_vec
-	for (std::size_t k(0); k<_data_vec->size(); ++k) {
-		if ((*_data_vec)[k]->getID() != k) {
-			(*_data_vec)[k]->setID(k);
+	for (std::size_t k(1); k<rm_pos.size(); ++k) {
```

Comment:
Tom, would you please add a short comment for each for-loop what it is doing, like it is done in the line 82 below.